web/packages/teleport/src/Discover/SelectResource/SelectResource.tsx

```diff
@@ -140,7 +142,6 @@ export function SelectResource(props: SelectResourceProps) {
                 resourceCardProps = {
                   onClick: () => {
                     if (r.hasAccess) {
-                      props.onSelect(r);
```

Comment:
I had removed it since removing the `unguidedLink` caused this to error out due to not having a `viewConfig` since the `Application` flow is a dialog as opposed to a Discover flow. I have added a workaround.